book/embeds.md

```diff
@@ -608,147 +608,29 @@ aspect ratio accordingly. Otherwise the page layout will look bad and cause
-more complex widgets like embedded maps, videos, social media buttons, ads and
-so on? Especially since such use cases often come from some third-party source,
-it would be weird to try to mix all the code for an embedded ad, video or map
-directly into your web page (and weird for the company providing the widget to
-trust your site enough to do so). In these cases we need a technology
-that "embeds" an externally-rendered widget into a web page in a modularized,
-flexible and secure way, but *also* allows interactions that coordinate well
-with the embedding web page and browser features.
-
-There are two possible ways to achieve this:
-
-* External content that is "outside the web", meaning it's not HTML & CSS.
-
-* External content that is "inside the web".
-
-The first type is a *plugin*. There have been many attempts at plugins on the
-web over the years. Some provided a programming language and mechanism for
-interactive UI, such as [Java applets][java-applets] or [Flash].^[YouTube
-originally used Flash for videos, for exmaple.] Others
-provided a way to embed other content types into a web page, such as
-[PDF]. But plugins suffer from a lot of the same accessibility and other
-"platform integration" drawbacks of `<canvas>`, and also have to provide
- duplicate solutions to all of the UI rendering problems we've already solved
- for browsers in the first place.
-
-[java-applets]: https://en.wikipedia.org/wiki/Java_applet
-[Flash]: https://en.wikipedia.org/wiki/Adobe_Flash
-[PDF]: https://developer.mozilla.org/en-US/docs/Learn/HTML/Multimedia_and_embedding/Other_embedding_technologies#the_embed_and_object_elements
-
-So the web has a second approach that solves all of these problems at once: let
-the developer embed one web page inside another, via the `<iframe>` element. As
-you'll see, this approach neatly solves all of these problems---accessibility,
-input, etc come "for free". And iframes are a great way to
-include third-party content content.
+So far, our browser has two kinds of embedded content: images and
+input elements. While both are important and widely-used[^variants]
+they don't offer quite the customizability and flexibility[^openui]
+that more complex embedded content---like maps, PDFs, ads, and social
+media controls---requires. In modern browsers, these are handled by
+*embeding one webpage within another* using the `<iframe>` element.
+
+[^variants]: As are variations like the [`<canvas>`][canvas-elt]
+    element. Instead of loading an image from the network, JavaScript
+    can draw on a `<canvas>` element via an API. Unlike images,
+    `<canvas>` element's don't have intrinsic sizes, but besides that
+    they are pretty similar.
 
 [canvas-elt]: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/canvas
+    
+[^openui]: There's actually [ongoing work](https://open-ui.org/) on
+    [various][shadow-dom] [APIs][form-el] to allow web pages to
```

Comment:
This shortening lost some of the key meaning. OpenUI is not standardizing additional shadow DOM features, they are standardizing things like customizable select elements. Shadow DOM and form internals are tech already developed that these structures build upon.